preprocessing/science/read_ini_fault_parameter.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+import easi
+import seissolxdmf
+import seissolxdmfwriter as sxw
+import argparse
+import numpy as np
+
+
+class seissolxdmfExtended(seissolxdmf.seissolxdmf):
+    def __init__(self, xdmfFilename):
+        super().__init__(xdmfFilename)
+        self.xyz = self.ReadGeometry()
+        self.connect = self.ReadConnect()
+
+    def ReadFaultTags(self):
+        """Read partition array"""
+        return self.Read1dData("fault-tag", self.nElements, isInt=True).T
+
+    def ComputeCellCenters(self):
+        """compute cell center array"""
+        return (
+            self.xyz[self.connect[:, 0]]
+            + self.xyz[self.connect[:, 1]]
+            + self.xyz[self.connect[:, 2]]
+        ) / 3.0
+
+    def ComputeCellNormals(self, ref_vector):
+        un = np.cross(
+            self.xyz[self.connect[:, 1], :] - self.xyz[self.connect[:, 0], :],
+            self.xyz[self.connect[:, 2], :] - self.xyz[self.connect[:, 0], :],
+        )
+        norm = np.apply_along_axis(np.linalg.norm, 1, un)
+        un = un / norm[:, np.newaxis]
+        mysign = np.sign(np.dot(un, ref_vector))
+        un = un * mysign[:, np.newaxis]
+        return un
+
+
+def compute_tractions(dicStress, un):
+    # compute Strike and dip direction
+    us = np.zeros(un.shape)
+    us[:, 0] = -un[:, 1]
+    us[:, 1] = un[:, 0]
+    norm = np.apply_along_axis(np.linalg.norm, 1, us)
+    us = us / norm[:, np.newaxis]
+    ud = np.cross(un, us)
+
+    def compute_traction_vector(dicStress, un):
+        nel = un.shape[0]
+        Tractions = np.zeros((nel, 3))
+        a = np.stack((dicStress["s_xx"], dicStress["s_xy"], dicStress["s_xz"]), axis=1)
+        Tractions[:, 0] = np.sum(a * un, axis=1)
+        a = np.stack((dicStress["s_xy"], dicStress["s_yy"], dicStress["s_yz"]), axis=1)
+        Tractions[:, 1] = np.sum(a * un, axis=1)
+        a = np.stack((dicStress["s_xz"], dicStress["s_yz"], dicStress["s_zz"]), axis=1)
+        Tractions[:, 2] = np.sum(a * un, axis=1)
+        return Tractions
+
+    Tractions = compute_traction_vector(dicStress, un)
+
+    outDict = {}
+    # compute Traction components
+    outDict["T_n"] = np.sum(Tractions * un, axis=1)
+    outDict["T_s"] = np.sum(Tractions * us, axis=1)
+    outDict["T_d"] = np.sum(Tractions * ud, axis=1)
+    print(outDict["T_n"].shape)
+    return outDict
+
+
+# parsing python arguments
```

Comment:
Maybe put all of this into a dedicated `main` function? (and add a `if __name__ == '__main__': main()`)